TezosKit/Client/RPCResponseHandler.swift

```diff
@@ -0,0 +1,50 @@
+// Copyright Keefer Taylor, 2019
+
+import Foundation
+
+/**
+ * A response handler handles responses that are received when network requests are completed.
+ */
+public class RPCResponseHandler {
+  /** The queue that callbacks from requests will be made on. */
+  private let callbackQueue: DispatchQueue
+
+  /** Initialize a new response handler with the given callback queue. */
+  public init(callbackQueue: DispatchQueue) {
+    self.callbackQueue = callbackQueue
+  }
+
+  /** Handle the given response from making the given RPC. */
+  public func handleResponse<T>(rpc: TezosRPC<T>, data: Data?, response: URLResponse?, error: Error?) {
+    // Check if the response contained a 200 HTTP OK response. If not, then propagate an error.
+    if let httpResponse = response as? HTTPURLResponse,
```

Comment:
Would response ever be nil in a non-error situation?

Reply:
NSURLSession only guarantees a URLResponse object.

What else would you suggest here?

Reply:
Should the argument for this method be ,response: URLResponse, in that case.  Without the optional?